### PR TITLE
TF-1328 Quick archive message

### PR DESCRIPTION
**Pull request by hieutbui**
## Issue
---
#1328 

## Resolved
---
- Web

https://github.com/linagora/tmail-flutter/assets/80142234/fddf81dd-66c5-468c-9231-46de853f39aa

- Android

https://github.com/linagora/tmail-flutter/assets/80142234/263cb8c7-852e-4c8a-99c1-74190b410b22

### CODE REVIEW

**Comment by hoangdat** [comment]
@hieutbui 
- [ ] don't show number of archive folder

File: lib/features/mailbox/presentation/mailbox_controller.dart [hunk 1]
```diff
@@ -1346,4 +1347,25 @@ class MailboxController extends BaseMailboxController with MailboxActionHandlerM
...
+          {mailboxContain.id: [email.id!]},
+          archiveMailboxId,
+          MoveAction.moving,
+          EmailActionType.moveToMailbox,
+          destinationPath:findNodePath(archiveMailboxId)
+        );
+          mailboxDashBoardController.moveToMailbox(
+          mailboxDashBoardController.sessionCurrent!,
+          mailboxDashBoardController.accountId.value!,
+          moveToArchiveMailboxRequest
```
Comment (hoangdat) [inline lib/features/mailbox/presentation/mailbox_controller.dart:1366]: indent

File: lib/features/mailbox/presentation/mailbox_controller.dart [hunk 2]
```diff
@@ -1346,4 +1347,25 @@ class MailboxController extends BaseMailboxController with MailboxActionHandlerM
       mailboxDashBoardController.emptySpamFolderAction(spamFolderId: presentationMailbox.id);
     }
   }
+
+  void archiveMessage(PresentationEmail email) {
```
Comment (hoangdat) [inline lib/features/mailbox/presentation/mailbox_controller.dart:1351]: why not put it in `mailbox_dashboard_controller`?

File: lib/features/email/presentation/controller/single_email_controller.dart [hunk 3]
```diff
@@ -93,6 +94,7 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
 
   final mailboxDashBoardController = Get.find<MailboxDashBoardController>();
   final emailSupervisorController = Get.find<EmailSupervisorController>();
+  final mailboxController = Get.find<MailboxController>();
```
Comment (hoangdat) [inline lib/features/email/presentation/controller/single_email_controller.dart:97]: why `single_email_controller` need to know about `mailboxController`?

File: lib/features/thread/presentation/mixin/email_action_controller.dart [hunk 4]
```diff
@@ -35,6 +36,7 @@ import 'package:tmail_ui_user/main/utils/app_utils.dart';
 mixin EmailActionController {
 
   final mailboxDashBoardController = Get.find<MailboxDashBoardController>();
+  final mailboxController = Get.find<MailboxController>();
```
Comment (hoangdat) [inline lib/features/thread/presentation/mixin/email_action_controller.dart:39]: idem

**Review by hoangdat — APPROVED** [review]

**Comment by hoangdat** [comment]
please rebase

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart [hunk 5]
```diff
@@ -2394,6 +2394,28 @@ class MailboxDashBoardController extends ReloadableController {
     return false;
   }
 
+  void archiveMessage(PresentationEmail email) {
+    final mailboxContain = email.findMailboxContain(mapMailboxById);
+    if (mailboxContain != null) {
+      final archiveMailboxId = getMailboxIdByRole(PresentationMailbox.roleArchive);
+      final archiveMailboxPath = mapMailboxById[archiveMailboxId]?.getDisplayName(currentOverlayContext!);
```
Comment (dab246) [inline lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:2401]: Please use `context` from the widget.

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart [hunk 6]
```diff
@@ -2394,6 +2394,28 @@ class MailboxDashBoardController extends ReloadableController {
...
+      if (archiveMailboxId != null) {
+        final moveToArchiveMailboxRequest = MoveToMailboxRequest(
+          {mailboxContain.id: [email.id!]},
+          archiveMailboxId,
+          MoveAction.moving,
+          EmailActionType.moveToMailbox,
+          destinationPath:archiveMailboxPath
```
Comment (dab246) [inline lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:2408]:
```suggestion
          destinationPath: archiveMailboxPath
```

File: lib/features/thread/presentation/thread_controller.dart [hunk 7]
```diff
@@ -1196,7 +1198,16 @@ class ThreadController extends BaseController with EmailActionController {
...
+    return isInArchiveMailbox(email)
+      ? DismissDirection.startToEnd
+      : DismissDirection.horizontal;
+  }
+
+  bool isInArchiveMailbox(PresentationEmail email) {
+    if (email.mailboxContain != null) {
+      return email.mailboxContain!.isArchive == true;
+    }
+    return false;
```
Comment (dab246) [inline lib/features/thread/presentation/thread_controller.dart:1210]:
```suggestion
  bool isInArchiveMailbox(PresentationEmail email) => email.mailboxContain?.isArchive == true;
```

File: lib/features/thread/presentation/thread_view.dart [hunk 8]
```diff
@@ -463,6 +467,36 @@ class ThreadView extends GetWidget<ThreadController>
...
+                  AppLocalizations.of(context).archiveMessage,
+                  style: const TextStyle(
+                    fontSize: 15,
+                    color: AppColor.primaryColor,
+                  ),
+                ),
+              ],
+            ),
+          ),
+        )
```
Comment (dab246) [inline lib/features/thread/presentation/thread_view.dart:498]: indent

File: lib/features/thread/presentation/thread_view.dart [hunk 9]
```diff
@@ -662,13 +698,39 @@ class ThreadView extends GetWidget<ThreadController>
...
+          fit: BoxFit.fill,
+          colorFilter: AppColor.colorTextButton.asFilter()
+        ),
+        AppLocalizations.of(context).archiveMessage,
+        email,
+        iconLeftPadding: controller.responsiveUtils.isMobile(context)
+          ? const EdgeInsets.only(left: 12, right: 16)
+          : const EdgeInsets.only(right: 12),
+        iconRightPadding: controller.responsiveUtils.isMobile(context)
+          ? const EdgeInsets.only(right: 12)
```
Comment (dab246) [inline lib/features/thread/presentation/thread_view.dart:718]: Please use `EdgeInsetsDirectional`

**Comment by dab246** [comment]
@hieutbui  Please rebase it

**Review by dab246 — APPROVED** [review]
lgtm

**Pull request merged**